### PR TITLE
Add code for generic pluralization

### PR DESCRIPTION
**Pull request by fingolfin**
This is motivated by the changes in PR #1320

### CODE REVIEW

File: src/generic/QuotientModule.jl [hunk 1]
```diff
@@ -54,12 +54,7 @@ supermodule(M::QuotientModule{T}) where T <: RingElement = M.m
 ###############################################################################
 
 function show_gens_rels(io::IO, N::AbstractAlgebra.FPModule{T}) where T <: RingElement
-   print(io, " with ", ngens(N), " generator")
-   if ngens(N) == 1
-      print(io, " and ")
-   else
-      print(io, "s and ")
-   end
+   print(io, " with ", ItemQuantity(ngens(N), "generator"), " and ")
```
Comment (fingolfin) [inline src/generic/QuotientModule.jl:57]: This showcases how to use the new code.

File: src/Pluralize.jl [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+0 oxen
+
+julia> ItemQuantity(1, "ox", "oxen")
+1 ox
+
+julia> ItemQuantity(2, "ox", "oxen")
+2 oxen
+```
+"""
+struct ItemQuantity
```
Comment (fingolfin) [inline src/Pluralize.jl:93]: Happy to change this if someone has a better name suggestion, this one is, err, "stolen", from EnglishText.jl

**Review by thofma — APPROVED** [review]

**Comment by thofma** [comment]
This time the CI error is real.

**Comment by fingolfin** [comment]
Yeah I forgot to `export` the two new names.

**Comment by fingolfin** [comment]
I just exported them, let's see if that fixes all errors or if there are more things to be fixed.

That said, perhaps they shouldn't be exported? In that case I can also instead adjust the doctests.

**Comment by fingolfin** [comment]
@thofma if you are happy, feel free to merge, then we can use it in PR #1320

**Pull request merged**